src/test/java/com/google/jenkins/plugins/k8sengine/client/CloudResourceManagerClientTest.java

```diff
@@ -23,130 +23,131 @@
 import org.mockito.junit.MockitoJUnitRunner;
+import org.mockito.stubbing.Answer;
 
 @RunWith(MockitoJUnitRunner.class)
 public class CloudResourceManagerClientTest {
-  private static final String TEST_CREDENTIALS_ID = "test-credentials";
+  private static final String SORTED_CREDENTIALS_ID = "sorted-credentials";
+  private static final String UNSORTED_CREDENTIALS_ID = "unsorted-credentials";
+  private static final String EMPTY_CREDENTIALS_ID = "empty-credentials";
   private static final String ERROR_CREDENTIALS_ID = "error-credentials";
+  private static final String NULL_CREDENTIALS_ID = "null-credentials";
   private static final List<String> TEST_PROJECT_IDS_SORTED =
       Arrays.asList("test-project-id-a1", "test-project-id-a2", "test-project-id-a3");
   private static final List<String> TEST_PROJECT_IDS_UNSORTED =
       Arrays.asList("test-project-id-b4", "test-project-id-b2", "test-project-id-b5");
 
-  private static List<Project> listOfProjects;
   private static Jenkins jenkins;
   private static ClientFactory clientFactorySpy;
 
   @BeforeClass
   public static void init() throws IOException {
-    listOfProjects = new ArrayList<>();
-
-    CloudResourceManager cloudResourceManager = Mockito.mock(CloudResourceManager.class);
-    Projects projects = Mockito.mock(Projects.class);
-    Projects.List listProjects = Mockito.mock(Projects.List.class);
-    ListProjectsResponse listProjectsResponse = Mockito.mock(ListProjectsResponse.class);
-    Mockito.when(cloudResourceManager.projects()).thenReturn(projects);
-    Mockito.when(projects.list()).thenReturn(listProjects);
-    Mockito.when(listProjects.execute()).thenReturn(listProjectsResponse);
-    Mockito.when(listProjectsResponse.getProjects()).thenReturn(listOfProjects);
-    CloudResourceManagerClient client = new CloudResourceManagerClient(cloudResourceManager);
-
-    CloudResourceManager errorCloudResourceManager = Mockito.mock(CloudResourceManager.class);
-    Projects errorProjects = Mockito.mock(Projects.class);
-    Projects.List errorListProjects = Mockito.mock(Projects.List.class);
-    Mockito.when(errorCloudResourceManager.projects()).thenReturn(errorProjects);
-    Mockito.when(errorProjects.list()).thenReturn(errorListProjects);
-    Mockito.when(errorListProjects.execute()).thenThrow(new IOException());
-    CloudResourceManagerClient errorClient =
-        new CloudResourceManagerClient(errorCloudResourceManager);
-
-    // Credentials are passed in at the time of creation.
     jenkins = Mockito.mock(Jenkins.class);
     clientFactorySpy = Mockito.spy(ClientFactory.class);
-    ClientFactory clientFactory = Mockito.mock(ClientFactory.class);
-    ClientFactory errorClientFactory = Mockito.mock(ClientFactory.class);
-    Mockito.when(clientFactory.cloudResourceManagerClient()).thenReturn(client);
-    Mockito.when(errorClientFactory.cloudResourceManagerClient()).thenReturn(errorClient);
-    Mockito.doReturn(clientFactory)
+    Mockito.doAnswer(
```

Comment:
Same feedback as the KubernetesEngineBuilderTest's init() method.

Reply:
Done